src/sql/data.d.ts

```diff
@@ -71,6 +71,43 @@ declare module 'data' {
 		export function registerProvider(provider: SerializationProvider): vscode.Disposable;
 	}
 
+	/**
+	 * Namespace for connection management
+	 */
+	export namespace connection {
+		/**
+		 * Get the current connection based on the active editor or Object Explorer selection
+		*/
+		export function getCurrentConnection(): Thenable<Connection>;
+
+		/**
+		 * Get all connected connections
```

Comment:
nit: "connected connections" sounds a little redundant (though I know what you mean).  Maybe "active connections" is better.